spring-observability-core/src/main/java/org/springframework/observability/micrometer/listener/MicrometerLongRunningTaskRecordingListener.java

```
@@ -0,0 +1,106 @@
+import io.micrometer.core.instrument.Tag;
+
+import org.springframework.observability.event.Recording;
+import org.springframework.observability.event.instant.InstantRecording;
+import org.springframework.observability.event.interval.IntervalLongRunningHttpServerEvent;
+import org.springframework.observability.event.interval.IntervalRecording;
+import org.springframework.observability.event.listener.RecordingListener;
+import org.springframework.observability.event.tag.Cardinality;
+
+/**
+ * {@link RecordingListener} that uses Micrometer's API to record long running tasks.
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.0.0
+ */
+public class MicrometerLongRunningTaskRecordingListener
+		implements MetricsRecordingListener<MicrometerLongRunningTaskRecordingListener.LongRunningTaskContext> {
+
+	private final MeterRegistry registry;
+
+	/**
+	 * @param registry The registry to use to record events.
+	 */
+	public MicrometerLongRunningTaskRecordingListener(MeterRegistry registry) {
+		this.registry = registry;
+	}
+
+	@Override
+	public boolean isApplicable(Recording<?, ?> recording) {
+		return recording.getEvent() instanceof IntervalLongRunningHttpServerEvent;
+	}
+
+	@Override
+	public void onStart(IntervalRecording<LongRunningTaskContext> intervalRecording) {
+		LongTaskTimer.Sample sample = LongTaskTimer.builder(intervalRecording.getEvent().getLowCardinalityName())
+				.description(intervalRecording.getEvent().getDescription()).tags(toTags(intervalRecording))
+				.register(this.registry).start();
+		intervalRecording.getContext().addSample(sample);
+	}
+
+	@Override
+	public void onStop(IntervalRecording<LongRunningTaskContext> intervalRecording) {
+		intervalRecording.getContext().getSamples().forEach(LongTaskTimer.Sample::stop);
+	}
+
+	@Override
+	public void onError(IntervalRecording<LongRunningTaskContext> intervalRecording) {
+		// TODO: If error add a tag
+	}
+
+	@Override
+	public void record(InstantRecording instantRecording) {
+
```

I would throw an `UnsupportedOperationException` since if this happens that's most probably a programming error (the listener should only be active for interval recordings).

---

We could break production code if we do it